### PR TITLE
Initial LogAnalyticEventRuleAction

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-8296

## Changes
- Add LogAnalyticEventRuleAction

### CODE REVIEW

File: src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        agency.submit(x, new ContextAgent() {
+          @Override
+          public void execute(X x) {
+            var event = new AnalyticEvent();
+            event.setName(getEventName());
+            event.setTimestamp(new Date());
```
Comment (jlhughes) [inline src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js:44]: Is this an after rule?  Thinking if beneficial if the timestamp aligns with the objects created or lastmodified timestamp.

Comment (lchanmann) [reply src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js:44]: This is a rule action can be on either before or after rule. Would like to use for after rule in https://github.com/nanoPayinc/NANOPAY/pull/15547 so the id will be generated and can be used as the traceId of the analytic event.

We can consider using object created/lastModified as the timestamp but the difference would just be some milliseconds and we will need to handle if the object doesn't have created or lastModified based on the rule operation.

Comment (jlhughes) [reply src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js:44]: we'll lets do that: if lastModified > 0 else created

Comment (jlhughes) [reply src/foam/nanos/analytics/ruler/LogAnalyticEventRuleAction.js:44]: Forget my last comment.  the timestamp as is fine.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**